eth2/state_processing/src/per_block_processing/block_signature_verifier.rs

```diff
@@ -46,23 +47,34 @@ impl From<BlockOperationError<AttestationInvalid>> for Error {
 ///
 /// This allows for optimizations related to batch BLS operations (see the
 /// `Self::verify_entire_block(..)` function).
-pub struct BlockSignatureVerifier<'a, T: EthSpec> {
+pub struct BlockSignatureVerifier<'a, T, F>
+where
+    T: EthSpec,
+    F: Fn(usize) -> Option<Cow<'a, G1Point>> + Clone,
+{
     block: &'a SignedBeaconBlock<T>,
+    get_pubkey: Box<F>,
```

Comment:
This compiles without the `Box`, do we need it?

Reply:
Removed, thank you!